packages/http/src/validator/validators/params.ts

```diff
@@ -1,22 +1,25 @@
-import { DiagnosticSeverity, HttpParamStyles, IHttpParam } from '@stoplight/types';
-import { upperFirst } from 'lodash';
+import { DiagnosticSeverity, Dictionary, HttpParamStyles, IHttpParam } from '@stoplight/types';
+import { mapKeys, upperFirst } from 'lodash';
 
 import { IPrismDiagnostic } from '@stoplight/prism-core/src/types';
 import { IHttpParamDeserializerRegistry } from '../deserializers/types';
 import { IHttpValidator } from './types';
 import { validateAgainstSchema } from './utils';
 
-export class HttpParamsValidator<Target, Spec extends IHttpParam> implements IHttpValidator<Target, Spec> {
+export class HttpParamsValidator<Target extends Dictionary<unknown, string>, Spec extends IHttpParam>
+  implements IHttpValidator<Target, Spec> {
   constructor(
     private _registry: IHttpParamDeserializerRegistry<Target>,
     private _prefix: string,
     private _style: HttpParamStyles,
   ) {}
 
   public validate(target: Target, specs: Spec[]): IPrismDiagnostic[] {
+    const loweredCaseTarget = mapKeys(target, (_value, key) => key.toLowerCase());
+
     const { _registry: registry, _prefix: prefix, _style: style } = this;
     return specs.reduce<IPrismDiagnostic[]>((results, spec) => {
-      if (!(spec.name in target) && spec.required === true) {
+      if (!(spec.name in loweredCaseTarget) && spec.required === true) {
```

Comment:
I think we need to do this for https://github.com/stoplightio/prism/blob/451a5a43db85cfcf62a49fabd151fdf5385143bc/packages/http/src/validator/validators/params.ts#L35 too